presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -771,6 +771,42 @@ public void testSchemaAuthorization()
         assertUpdate(admin, "DROP ROLE admin");
     }
 
+    @Test
+    public void testTableAuthorization()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("hive").withRole("hive", new SelectedRole(ROLE, Optional.of("admin"))).build())
+                .build();
+
+        Session alice = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("alice").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        Session bob = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("bob").withPrincipal(getSession().getIdentity().getPrincipal()).build())
+                .build();
+
+        assertUpdate(admin, "CREATE SCHEMA test_table_authorization");
+        assertUpdate(admin, "CREATE ROLE admin");
+
+        assertUpdate(admin, "CREATE TABLE test_table_authorization.foo (col int)");
+        assertUpdate(admin, "ALTER TABLE test_table_authorization.foo SET AUTHORIZATION alice");
+        assertQueryFails(bob,
```

Comment:
assertAccessDenied? Then you don't need two users.

Please put `bob` to separate line